mne_connectivity/spectral/time.py

```diff
@@ -67,9 +85,9 @@ def spectral_connectivity_time(data, names=None, method='coh', indices=None,
     mt_bandwidth : float | None
         The bandwidth of the multitaper windowing function in Hz.
         Only used in 'multitaper' mode.
-    freqs : array
-        Array of frequencies of interest for use in time-frequency
-        decomposition method (specified by ``mode``).
+    cwt_freqs : array
+        Array of frequencies of interest for time-frequency decomposition.
+        Only used in 'cwt_morlet' mode.
```

Comment:
`_compute_freq_mask` from `mne_connectivity.spectral.epochs.py` is used, and hence the frequencies outside the range specified by `fmin` and `fmax` are not used.

Reply:
Fine, but this behavior is not clear from the documentation.Hence why I wondered if this should be two different functions to avoid such confusion,.

Reply:
Yeah, it might be a good idea to break this up into two wrapper functions and make the existing function private. How would you go about naming such new functions? If this is done, then I believe `spectral_connectivity_epochs` should also be broken up, which comes with the cost of breaking the code of everyone currently using it.

Reply:
Mh fair point. Alternatively we could introduce a methods parameter and then be very clear in the documentation of which other parameters belong to which method. And throw clear error messages if "illegal" combinations are used. Thoughts, @adam2392, and maybe @larsoner ?

Reply:
I'm in favor of breaking up the functions. I think we can introduce a deprecation cycle for `spectral_connectivity_epochs`, but for `spectral_connectivity_time`, we could maybe do:

`spectral_wavelet_connectivity_time` and `spectral_fourier_connectivity_time`?

Reply:
have a look at how it was recently done for the new spectrum class in MNE-Python, which is created by the `compute_psd` methods. Basically, there are a bunch of shared params, then a `**method_kw` param for the 4 or 5 extra params exposed for each of the supported methods (in that case welch vs multitaper, but should work the same for wavelet and fourier here, I'd think).

Docstring: https://mne.tools/dev/generated/mne.io.Raw.html#mne.io.Raw.compute_psd

Implementation: 

https://github.com/mne-tools/mne-python/blob/08c8cad249e6e180096b0c8c3e9075ca6ae31bd6/mne/time_frequency/spectrum.py#L257-L277

Reply:
(in other words, I'm somewhat in favor of keeping this as one function, if the only reason for splitting it is that you want to expose a few non-overlapping params from each of the two methods)

Reply:
This is definitely one option. I don't have a strong opinion here, what about @adam2392, @britta-wstnr ?

Reply:
Okay for changing then to method_kwargs to keep in line with mne-Python!

Reply:
I looked at going for a `**method_kw` argument like in the case of psd above. However, I think this only adds confusion, as some of the method arguments are required, not optional as you would expect from additional keyword arguments. Some of the method parameters are also shared, and both shared and non-shared parameters are also used in the `spectral_connectivity_time` function itself.

Reply:
Ah in that case 2 separate functions is probably the right choice then